meson.build

```diff
@@ -0,0 +1,246 @@
+project('libass', 'c', license: 'ISC',
+        meson_version: '>= 0.55.0',
+        # cpp_std specified because meson compiler options are currently global
+        default_options: ['c_std=c99', 'cpp_std=c++11', 'buildtype=debugoptimized', 'warning_level=2'],
+        version: '0.15.0')
+
+conf = configuration_data()
+deps = []
+
+# Compiler setup
+
+cc = meson.get_compiler('c')
+
+if cc.get_argument_syntax() == 'gcc'
+    cflags = [
+        '-Wno-sign-compare',
+        '-Wno-unused-parameter',
+        '-Werror-implicit-function-declaration',
+        '-Wstrict-prototypes',
+        '-Wpointer-arith',
+        '-Wredundant-decls',
+        '-Wno-missing-field-initializers'
+    ]
+    add_project_arguments(cc.get_supported_arguments(cflags), language: 'c')
+endif
+
+test_args = []
+
+if host_machine.system() == 'linux'
+    test_args += '-D_GNU_SOURCE'
+    add_project_arguments('-D_GNU_SOURCE', language: 'c')
+elif host_machine.system() == 'darwin'
+    test_args += '-D_DARWIN_C_SOURCE'
+    add_project_arguments('-D_DARWIN_C_SOURCE', language: 'c')
+elif cc.get_id() != 'msvc'
+    test_args += '-D_POSIX_C_SOURCE=200809L'
+    add_project_arguments('-D_POSIX_C_SOURCE=200809L', language: 'c')
+endif
+
+# Configuration
+
+if get_option('large-tiles')
+    conf.set('CONFIG_LARGE_TILES', 1)
+endif
+
+conf.set('PACKAGE_NAME', 'libass')
+conf.set('PACKAGE_VERSION', meson.project_version())
+
+conf.set('CONFIG_SOURCEVERSION', '"meson, on or after: ' + meson.project_version() + '"')
+
+str_check_functions = [
+    'strdup',
+    'strndup'
+]
+
+foreach name: str_check_functions
+    if cc.has_function(name) and cc.has_header_symbol('string.h', name, args: test_args)
+        conf.set('HAVE_@0@'.format(name.to_upper()), 1)
+    endif
+endforeach
+
+# Dependencies
+
+m_dep = cc.find_library('m', required: false)
+if m_dep.found()
+    deps += m_dep
```

Comment:
For the record, all of these dep checks can just be

```
deps += cc.find_library('m', required: false)
```

and similar. If the dependency is a not-found-dependency then it will just do nothing.

Reply:
I think there was some reason I didn't do that originally... I wrote this port a couple years ago (and it's still not merged D:), so maybe the situation has changed?